# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/other/DocAccessibleChild.cpp
# Commit: e807c6507634
# Full Hash: e807c6507634bdeca647372d8815570d7750b3b2
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-04-02 09:44:13
# Regressor Bug: 1735970
# File Overlap Count: 1
# Description:
#   Bug 1735970 part 15: Support TableAccessibleBase and TableCellAccessibleBase for ATK. r=morgan
#   
#   We can use the base classes for both local and cached remote Accessibles.
#   However, non-cached RemoteAccessibles need to be handled separately still because we can't implement the table interfaces for those.
#   
# ==============================================================================

diff -r 03e98595f217 -r e807c6507634 accessible/ipc/other/DocAccessibleChild.cpp
--- a/accessible/ipc/other/DocAccessibleChild.cpp	Fri Apr 01 09:50:01 2022 +0000
+++ b/accessible/ipc/other/DocAccessibleChild.cpp	Fri Apr 01 09:50:01 2022 +0000
@@ -1204,9 +1204,9 @@
 #ifdef MOZ_ACCESSIBILITY_ATK
   TableAccessible* acc = IdToTableAccessible(aID);
   if (acc) {
-    LocalAccessible* header = AccessibleWrap::GetColumnHeader(acc, aCol);
+    Accessible* header = AccessibleWrap::GetColumnHeader(acc, aCol);
     if (header) {
-      *aHeader = reinterpret_cast<uint64_t>(header->UniqueID());
+      *aHeader = header->ID();
       *aOk = true;
     }
   }
@@ -1223,9 +1223,9 @@
 #ifdef MOZ_ACCESSIBILITY_ATK
   TableAccessible* acc = IdToTableAccessible(aID);
   if (acc) {
-    LocalAccessible* header = AccessibleWrap::GetRowHeader(acc, aRow);
+    Accessible* header = AccessibleWrap::GetRowHeader(acc, aRow);
     if (header) {
-      *aHeader = reinterpret_cast<uint64_t>(header->UniqueID());
+      *aHeader = header->ID();
       *aOk = true;
     }
   }
